### PR TITLE
[DATA-827] add GetTagsByFilter to data proto

### PR DESCRIPTION
**Pull request by alexis-wei**

### CODE REVIEW

File: proto/viam/app/data/v1/data.proto [hunk 1]
```diff
@@ -39,6 +39,9 @@ service DataService {
 
   // RemoveTagsToBinaryDataByFilter removes string tags from binary data based on the given filter.
   rpc RemoveTagsFromBinaryDataByFilter(RemoveTagsFromBinaryDataByFilterRequest) returns (RemoveTagsFromBinaryDataByFilterResponse);
+
+  // TagsByFilter gets all tags that match the given filter.
```
Comment (tahiyasalam) [inline proto/viam/app/data/v1/data.proto:43]: [nit]: "gets all unique tags" to indicate there will be no duplicates in the returned tags (IIRC)

**Review by tahiyasalam — CHANGES_REQUESTED** [review]
I left a small comment about language on the comments, but otherwise makes sense to me! Saving approval until the end, since it will automatically dismiss approval and re-request if you push any changes

File: proto/viam/app/data/v1/data.proto [hunk 2]
```diff
@@ -256,3 +259,13 @@ message RemoveTagsFromBinaryDataByFilterRequest {
 message RemoveTagsFromBinaryDataByFilterResponse {
   uint64 deleted_count = 1;
 }
+
+// TagsByFilterRequest requests the tags that exist for the given filter
+message TagsByFilterRequest {
```
[the remark this answers is not in the document]
Comment (alexis-wei) [reply proto/viam/app/data/v1/data.proto:264]: updated!

File: proto/viam/app/data/v1/data.proto (hunk 1 above)
Comment (alexis-wei) [reply proto/viam/app/data/v1/data.proto]: updated!

**Review by agreenb — APPROVED** [review]
💯

**Comment by agreenb** [comment]
> After you've finalized, I would suggest bumping the api versions of RDK, and then APP with this PR to ensure the API changes are consistent across all repos.

Thanks for adding the reminder @tahiyasalam! No need to bump the api version of RDK for this since it's a new nonbreaking change that's only used in app. Updating an existing field/method that's used in RDK (even just to be marked as `deprecated` or deleting a previous field) would require bumping RDK as well.

**Pull request merged**